### DISCUSSION TITLE
SUN2000L not SUN2000

### DISCUSSION THREAD

**Discussion by shortgame11** [wlcrs/huawei_solar#346] [Setting up the integration] [answered]
Hi,
I have a SUN2000L which I think is the older variant. Our setup uses a power meter and this was wired to pin 1 and 2 of the com port as that was the way these older inverters were meant to be setup. My problem is that whenever the EW11 is on, regardless of what combination of ports I use (Either power meter in 1 and 2 or 3 and 4 and EW11 the alternate) the power meter is unable to be communicated with (the app says there is a communication error) and I cannot connect the integration to the EW11. 
As soon as I turn off the EW11 the power meter can be found and communicated with fine, with it in either set of ports.

Any help would be appreciated.

**Comment by wlcrs** [comment, marked as answer]
I compared the [SUN2000L](https://support.huawei.com/enterprise/en/doc/EDOC1100011912/f77e4882/installing-the-signal-cable) and [SUN2000](https://support.huawei.com/enterprise/en/doc/EDOC1100136173/daff17bb/optional-connecting-signal-cables) manual wrt the "signal cable" installation:

*SUN2000L*
![SUN2000L](https://user-images.githubusercontent.com/2150060/229543215-c5cd321c-bbfc-42c9-9090-aba1c261694f.png)

*SUN2000*
![SUN2000](https://user-images.githubusercontent.com/2150060/229543356-c82129e2-eac8-4590-a8f7-f19bfebe0256.png)

As you can see, the connector behaves quite differently. I'm afraid that a serial connection to a SUN2000L is not possible. You could contact Huawei support to verify this.

**Reply by shortgame11** [reply]
I was under the impression that some people had it working. For instance:

https://community.home-assistant.io/t/integration-solar-inverter-huawei-2000l/132350/2513?u=shortgame11

**Reply by shortgame11** [reply]
Why did you mark this answered? The connector behaves moderately differently and you haven't addressed why other people believe they have it working. Once addressed then maybe the issue can be considered answered (By me, the person who asked the question). Thank you

**Reply by wlcrs** [reply]
Because I don't know how the answer can be more clear: the manual clearly shows that you cannot connect via serial to a SUN2000L. Your experience matches that. A random person on the internet who probably mindlessly added the 'L' to the type number doesn't change that.

So either you contact Huawei for a new insight, or my answer above will stand as the final and correct answer. (by me, the maintainer of this library who spent 10 minutes of his life yesterday -free of charge- researching the correct answer to your question, only to receive a thankless and aggressive answer in return). Thank you
